FIX-#337: Fix MpiHosts parameter usage (#378)
modin-project/unidist: docs/using_unidist/unidist_on_mpi.rst

Unidist on MPI cluster
""""""""""""""""""""""

> Regardless of the chosen usage model (SPMD model or Controller/Worker model), there are two options for running on a cluster

Reviewer: ```suggestion
Regardless of the chosen usage model (SPMD model or Controller/Worker model),
there are two options for running on a cluster.
```

> Running with `mpiexec` command
> ------------------------------

Reviewer: ```suggestion
Running with ``mpiexec`` command
--------------------------------
```

This option is the most preferred and customizable.

> Running is almost the same as in a single node, but you should use the appropriate parameter for "mpiexec". This parameter differs depending on the mpi implementation used.

Reviewer: ```suggestion
Running is almost the same as in a single node, but, in addition,
you should use the appropriate parameter for ``mpiexec``.
This parameter differs depending on the MPI implementation used.
```

> For Intel MPI or MPICH: `-hosts host1,host2`. You also can see 

Reviewer: ```suggestion
* For Intel MPI or MPICH: ``-hosts host1,host2``. You can also see 
```
For MPICH it is `-host`, isn't it? https://www.mpich.org/static/docs/v3.4.x/www1/mpiexec.html#:~:text=processes%20to%20use-,%2Dhost%20%3Chostname%3E,-Name%20of%20host

> `Controlling Process Placement with the Intel® MPI Library <https://www.intel.com/content/www/us/en/developer/articles/technical/controlling-process-placement-with-the-intel-mpi-library.html>` 

Reviewer: Correct the link.

Author: fixed

> for more deeper customize. 

Reviewer: ```suggestion
for deeper customization.
```

> For OpenMPi: `-host host1:n1,...,hostM:nM`

Reviewer: ```suggestion
* For OpenMPI: ``-host host1:n1,...,hostM:nM``
```

where n1, ..., nM is the number of processes on each node, including system processes.

> You also can see `Scheduling processes across hosts with OpenMPI Library <https://docs.open-mpi.org/en/v5.0.x/launching-apps/scheduling.html>` for more deeper customize. 

Reviewer: ```suggestion
You can also see `Scheduling processes across hosts with OpenMPI Library <https://docs.open-mpi.org/en/v5.0.x/launching-apps/scheduling.html>` for deeper customization. 
```

> Running without 'mpiexec' command
> ---------------------------------

Reviewer: ```suggestion
Running without ``mpiexec`` command
-----------------------------------
```

> To run the unidist on a cluster without the `mpiexec` command, you should specify hosts to run on.

Reviewer: ```suggestion
To run unidist on MPI in a cluster without the ``mpiexec`` command, you should specify hosts to run on.
```

Unidist on MPI cluster
""""""""""""""""""""""

> Regardless of the chosen usage model (SPMD model or Controller/Worker model),

Reviewer: ```suggestion
Regardless of the chosen execution model (SPMD or Controller/Worker),
```

there are two options for running on a cluster.

Running with ``mpiexec`` command
--------------------------------

This option is the most preferred and customizable.

Running is almost the same as in a single node, but, in addition,
you should use the appropriate parameter for ``mpiexec``.
This parameter differs depending on the MPI implementation used.

* For Intel MPI or MPICH: ``-hosts host1,host2``. You can also see 
`Controlling Process Placement with the Intel® MPI Library <https://www.intel.com/content/www/us/en/developer/articles/technical/controlling-process-placement-with-the-intel-mpi-library.html>`_ 

> or `MPICH wiki <https://github.com/pmodels/mpich/blob/main/doc/wiki/how_to/Using_the_Hydra_Process_Manager.md>`_ for deeper customization.

Reviewer: Change the way for inserting links. To be discussed offline.

Author: done

* For OpenMPI: ``-host host1:n1,...,hostM:nM``

> where n1, ..., nM is the number of processes on each node, including system processes.

Reviewer: Which system processes do you mean?

Author: Sorry, I meant Unidist service processes  (root and one or some monitors). It has fixed.

You can also see `Scheduling processes across hosts with OpenMPI Library <https://docs.open-mpi.org/en/v5.0.x/launching-apps/scheduling.html>`_ for deeper customization. 

> 

Reviewer: ```suggestion
```

Running without ``mpiexec`` command
-----------------------------------

> To run unidist on MPI in a cluster without the ``mpiexec`` command, you should specify hosts to run on.

Reviewer: ```suggestion
To run unidist on MPI in a cluster without ``mpiexec`` command,
you should specify MPI hosts to run on using unidist configuration settings.
```

Running is the same as in a single node.

.. note::
    Root proccess will allways be executed locally and other proccesses will be spawned in order on the specified hosts.
    If you want to run root proccess on anoother host you should use `ssh host` before your command and thoroughly check that environment will be correct. 
    You can set some variables into ssh or activate conda envirenment before running python script.
    
    If you want to start a root process on another host, you should use `ssh host` before the command and carefully check that the environment is correct. 

>     You can set some variables in ssh command or activate the conda environment right before running the Python script:

Reviewer: Duplication.

Author: fixed

    If you want to run root proccess on anoother host you should use `ssh host` before your command and thoroughly check that environment will be correct. 
    You can set some variables into ssh or activate conda envirenment before running python script.
    
    If you want to start a root process on another host, you should use `ssh host` before the command and carefully check that the environment is correct. 
    You can set some variables in ssh command or activate the conda environment right before running the Python script:

.. code-block:: bash

>     $ export UNIDIST_IS_MPI_SPAWN_WORKERS=False

Reviewer: Why is everything below removed?

Author: because the structure has changed and there is no need to write about the SPMD model here

Running is the same as in a single node.

.. note::
    Root proccess will allways be executed locally and other proccesses will be spawned in order on the specified hosts.

>     If you want to run root proccess on anoother host, you should use `ssh host` before the command and carefully check that the environment is correct. 

Reviewer: ```suggestion
    If you want to run root process on a remote host, you should use `ssh host` before the command and carefully check that the environment is correct. 
```

    You can set some variables in ssh command or activate the conda envirenment right before running the Python script:

.. code-block:: bash

>     ssh host ENV_VARIABLE=value "source /PATH_TO_CONDA/activate CONDA_ENV; cd $PWD; python script.py"

Reviewer: ```suggestion
    ssh host ENV_VARIABLE=value "source $CONDA_PATH/bin/activate $CONDA_ENV; cd $PWD; python script.py"
```

Author: done
